test/Regex/control_character_escapes.js

```diff
@@ -0,0 +1,201 @@
+            matchRegExp("\\\\", re, null);
+            matchRegExp("////", re, null);
+            matchRegExp("cfG6", re, null);
+            
+            re = /\cG\cf/; //'G' = ascii x47, 'f' = ascii x66
+            matchRegExp("\x00\x03\x07\x06\x16\x07\x08", re, "\x07\x06");
+            matchRegExp("\\", re, null);
+            matchRegExp("/", re, null);
+            matchRegExp("\\cG\\c6\\cf", re, null);
+            
+            re = /[\cz\cZ]+/; //'z' = ascii x7A, 'Z' = ascii x5A, have the same lowest 5 bits
+            matchRegExp("\x00\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0a\x0b\x0c\x0d\x0e\x0f" + 
+                        "\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f", re, "\x1a");
+            matchRegExp("\\\\", re, null);
+            matchRegExp("////", re, null);
+            matchRegExp("ccczZ", re, null);
+        }
+    },
+    {
+        name : "Control characters followed by a non-word character ([^A-Za-z0-9_])",
+        body : function () 
+        {
+            re = /[\c*]+/; //'*' = ascii 42
+            matchRegExp("\x0a\x09\x08", re, null);
+            matchRegExp("a*c*b*d*", re, "*c*");
+            matchRegExp("\\\\", re, "\\\\");
+            matchRegExp("////", re, null);
+            matchRegExp("ccc", re, "ccc");
+            
+            re = /[\c}]*/; //'}' = ascii 125
+            matchRegExp("\x1d\x7d\x3d", re, "");
+            matchRegExp("}c}}cd*c*b*d*", re, "}c}}c");
+            matchRegExp("\\\\", re, "\\\\");
+            matchRegExp("////", re, "");
+            matchRegExp("ccc", re, "ccc");
+            
+            re = /[\c;]+/; //';' = ascii 59
+            matchRegExp("\x1b\x1c", re, null);
+            matchRegExp("d;c;d;*", re, ";c;");
+            matchRegExp("\\\\", re, "\\\\");
+            matchRegExp("////", re, null);
+            matchRegExp("ccc", re, "ccc");
+            
+            re = /\c%/; //'%' = ascii x25
+            matchRegExp("\\", re, null);
+            matchRegExp("\\", re, null);
+            matchRegExp("\\c%", re, "\\c%");
+            matchRegExp("\x05", re, null);
+        }
+    },
+    {
+        name : "Control Character tests with unicode flag present",
+        body : function () 
+        {
+            re = /[\cAg]+/u; //'A' = ascii x41
+            matchRegExp("abcdefghi", re, "g");
+            matchRegExp("\\\\", re, null);
+            matchRegExp("////", re, null);
+            matchRegExp("\x01\x01gg\x02\x04ggg", re, "\x01\x01gg");            
+            
+            re = /[\czA]+/u;  //'z' = ascii x7A
+            matchRegExp("abcdefghi", re, null);
+            matchRegExp("\\\\", re, null);
+            matchRegExp("////", re, null);
+            matchRegExp("YZA\x1aABC", re, "A\x1aA");    
+            
+            assert.throws(() => eval("\"\".match(/[\\c]/u)"), SyntaxError, "Expected an error due to non-letters being disallowed from control character when unicode flag present", 

```

Comment:
It's a bit hard when they're so similar, but different descriptions for the following will allow us to find the right one when the test fails.